runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java

```diff
@@ -476,16 +476,15 @@ public void translateNode(
     public void translateNode(
         PTransform<PCollection<KV<K, InputT>>, PCollection<KV<K, OutputT>>> transform,
         FlinkBatchTranslationContext context) {
-      DataSet<WindowedValue<KV<K, InputT>>> inputDataSet =
+      final DataSet<WindowedValue<KV<K, InputT>>> inputDataSet =
           context.getInputDataSet(context.getInput(transform));
-
-      CombineFnBase.GlobalCombineFn<InputT, AccumT, OutputT> combineFn =
-          ((Combine.PerKey) transform).getFn();
-
-      KvCoder<K, InputT> inputCoder = (KvCoder<K, InputT>) context.getInput(transform).getCoder();
-
-      Coder<AccumT> accumulatorCoder;
-
+      final Combine.PerKey<K, InputT, OutputT> combineTransform =
+          (Combine.PerKey<K, InputT, OutputT>) transform;
+      final CombineFnBase.GlobalCombineFn<InputT, AccumT, OutputT> combineFn =
+          (CombineFnBase.GlobalCombineFn<InputT, AccumT, OutputT>) combineTransform.getFn();
+      final KvCoder<K, InputT> inputCoder =
+          (KvCoder<K, InputT>) context.getInput(transform).getCoder();
+      final Coder<AccumT> accumulatorCoder;
```

Comment:
Code style: It would be great if refs were immutable by default, but adding final to everything I find adds verbosity with little added benefit.

Reply:
I agree that the benefit is discutable. 

Personally I find it helpful when it comes to long methods or trying to understand someone elses code, that you see that there would be no more assignents to the variable (if there is no final modifier, you know that there is something more going on).

Anyway, it's more of a personal taste. Do you want me to revert these?

Reply:
Yes, it depends on the situation. I use `final` mostly when there is a clear assign-once relationship with commonly used types. Since type-parameterized types are used here, there is very little chance of accidentally overwriting a ref. 

Just a discussion point. No need to revert.